Techqueueandfixes

twestura opened the pull request:
In progress changes for the new WololoKingdoms update.

Current changes:
* Fixes the House Attack Bug.
* Makes Demolition Ship attack stance default to Defensive when placed in the Scenario Editor.
* Makes Uprooting available only for Siege Onager civilizations.
* Fixes the Nomads house bug (important for DM games).
* Fixes the Slav team bonus when maps start with military buildings already present.
* Allows each unique technology to have its own unique icon.
* Applies the Aztec bonus to trebuchets instead of to the packed siege class.
* Separates ship death animations to allow for graphics in multiple directions.
* Adds various new units (Roaming Wolf, Transport Sharkatzor, and a Unit Spawner).
* Makes technologies queue-able (this PR contains the game data changes, still requires other exe changes).
* Duplicates death animations for ship slp files: Duplicate the animation frames for slps 495, 692, 1834, 2116, 2753, 2778, and 5175 eight times, so there are frames for a total of nine angles. And duplicate the animations frames for slp 441 four times, so there is a total of five angles.
* Modifies slp 50729 for unique techs: duplicate the 34th frame 31 times and the 108th frame 31 times.
* Incorporates https://github.com/twestura/WololoKingdoms/pull/1

twestura committed:
small whitespace change to test commit

twestura committed:
Correct swap of demoraft and demoship constants

twestura committed:
Update hero status of demo ship.

twestura committed:
Fix the house attack bug

twestura committed:
Remove unnecessary TODO statement

twestura committed:
Fix unit ids for demo ships, don't overwrite hero modes with their existing values

twestura committed:
Remove commented out code

twestura committed:
Add uprooting fix

twestura committed:
Add Nomads fix

twestura committed:
Fix slav team bonus when maps start with military buildings

twestura committed:
Update description string

twestura committed:
Be consistent with size_t instead of int for consts

twestura committed:
Split unique techs into individual icons

twestura committed:
Change how the aztec bonus applies to trebs.

twestura committed:
Update specification comment

twestura committed:
Separate ship death animations

twestura committed:
Add misc units

twestura committed:
Data changes for queue-able techs

goto-bus-stop commented:
let's get this in before #47, so that this update doesn't get blocked by merge conflicts

goto-bus-stop commented on libwololokingdoms/fixes/queuetechs.cpp:
```diff
@@ -0,0 +1,604 @@
+        break;
+      default:
+        assert(false, "Unit Class ID must be between 0 and 3, inclusive, to "
+                      "represent a resource cost.");
```
C-style `assert()` only takes a single argument, sometimes people do:
```c
    assert(!"Unit Class ID must be between 0 and 3, inclusive, to represent a resource cost.");
```
which is horrible but works.

This file also needs an `#include <cassert>` at the top to make the `assert()` macro available (it might be pulled in by something else in some standard library implementations, but it's not in my Clang version)

twestura replied:
Added the `include` statement.
I'll just leave that line as `assert(false);` and refactor the string to a comment.

goto-bus-stop committed:
Fix scenario editor crash because of missing name

goto-bus-stop committed:
Fix retrying an incomplete install (at least on Linux)

twestura committed:
Fix task list IDs for added wolf and sharkatzor

twestura committed:
Use single argument for c-style assert

twestura committed:
Edit comments for consistent style

twestura committed:
Create simple SLP frame duplication library

twestura committed:
Implement ship and tech icon slp modification

Jineapple committed:
- Fixes for directional sinking animations and unique tech icons
- Files in mod_override directory are properly indexed last.

goto-bus-stop committed:
Remove duplicate #include

goto-bus-stop committed:
Fix naming conflict

goto-bus-stop committed:
Initialize queue tech dummy units with correct IDs

twestura committed:
Merge pull request #1 from goto-bus-stop/techqueueandfixes

[WIP] Tweaks for the queueable tech exe

goto-bus-stop committed:
Add DLL mods to offline installer.

Jineapple committed:
Added methods to adjust the unit icon slp for tech queue icons automatically.

If icons are supposed to be modded (through the mod_override folder) the finished slp with tech icons included in the unit file has to be used. This could be adjusted/improved in the future.

Jineapple committed:
Interface fix for <768p resolutions

Jineapple committed:
Include Immobile Units AI and track the Script.AI folder

Jineapple committed:
Mad the installer window wider and arranged options in two columns. This solves the issue of the run button being cut off on displays with low resolution/high scaling.

Jineapple committed:
Switch the IDs between cutting and non-cutting onager so the regular/cutting one is in the mangonel-line for AIs

Jineapple committed:
Seperate the rest of the civs for IA/IU

Jineapple committed:
Rearranged civs in the history section

Instead of being sorted alphabetically within the expansion, it's now done by civ id. This is so civ jingles which seem to be played in order ID for the new civs fit together with the text.

Jineapple committed:
Changed faulty gate foundation armor

If general gate foundation armor is changed in future patch, this needs to be removed!

Jineapple committed:
Siege Tower fix

Jineapple committed:
Fixes wrong IDs for copied units causing problems

Adds some new language IDs for new units, some of these may still need changing

Jineapple committed:
Merge remote-tracking branch 'goto-bus-stop/install-mods' into techqueueandfixes

twestura committed:
Initial draft of balance changes

twestura committed:
Merge branch 'techqueueandfixes' of https://github.com/twestura/WololoKingdoms into techqueueandfixes

goto-bus-stop commented on libwololokingdoms/fixes/addunits.h:
```diff
@@ -0,0 +1,11 @@
+#ifndef ADDUNITS_H
+#define ADDUNITS_H
```
minor thing but we can just do `#pragma once` instead of the if guard dance

Jineapple committed:
Merge branch 'techqueueandfixes' of https://github.com/twestura/WololoKingdoms into techqueueandfixes

Jineapple committed:
Track AI files

Jineapple committed:
Merge branch 'bugfixing' into techqueueandfixes

Jineapple committed:
Use push_back as array is empty at init

Jineapple committed:
Create Beta 5.9 Version of the installer

Use #pragma once

Jineapple committed:
Symlink mmmods, use one basemodname everywhere

Jineapple committed:
Turkish language

Jineapple committed:
Update map def and gitignore

Jineapple committed:
Fix IA, use Beta Name

goto-bus-stop commented:
"Merged" into [`dev`](https://github.com/SiegeEngineers/WololoKingdoms/tree/dev)
